pkg/hub/custommetrics/custommetrics.go

```diff
@@ -0,0 +1,198 @@
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+)
+
+var (
+	//https://..../apis/cluster.open-cluster-management.io/v1/managedclusters
+	//correspond to
+	//.../apis/G/V/R
+	mcGVR = schema.GroupVersionResource{
+		Group:    "cluster.open-cluster-management.io",
+		Version:  "v1",
+		Resource: "managedclusters",
+	}
+
+	cvGVR = schema.GroupVersionResource{
+		Group:    "config.openshift.io",
+		Version:  "v1",
+		Resource: "clusterversions",
+	}
+
+	hubID = ""
+)
+
+//cluster_id = OCP ID of the Cluster (need to resolve for eks, etc)
+//type = K8s Distribution, e.g. OCP, EKS, etc
+//version = Distribution version
+//cluster_infrastructure_provider = value "Type" from cluster_infrastructure_provider
+//hub_id = cluster_id of hub server
+//cluster_name =User Display Name of Cluster (defaults to Id if not provided)
+var managedClusterMetric = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	Name: "a_managed_cluster",
+	Help: "Managed Cluster being managed by ACM Hub.",
+}, []string{"cluster_name", "cluster_id", "type", "version", "cluster_infrastructure_provider", "hub_id"})
+
+func getDynClient() (dynamic.Interface, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return dynamic.NewForConfig(config)
+}
+
+func getStaticData(c dynamic.Interface) {
+
+	cvObj, errCv := c.Resource(cvGVR).Get(context.TODO(), "version", metav1.GetOptions{})
+	if errCv != nil {
+		klog.Fatalf("Error getting cluster version %v \n", errCv)
+		panic(errCv.Error())
+	}
+	cv := &ocinfrav1.ClusterVersion{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(cvObj.UnstructuredContent(), &cv)
+	if err != nil {
+		klog.Fatalf("Error unmarshal cluster version object%v \n", err)
+		panic(errCv.Error())
+	}
+	hubID = string(cv.Spec.ClusterID)
+	klog.Infof("Hub id of this ACM Server is " + hubID)
+
+}
+
+func addCluster(obj interface{}) {
+	klog.Info("Adding Managed Clusters ####################")
+	j, err := json.Marshal(obj.(*unstructured.Unstructured))
+	if err != nil {
+		klog.Warning("Error on ManagedCluster marshal.")
+	}
+	managedCluster := clusterv1.ManagedCluster{}
+	err = json.Unmarshal(j, &managedCluster)
+	if err != nil {
+		klog.Warning("Error on ManagedCluster unmarshal.")
+
+	}
+
+	klog.Infof("Managed Cluster name being added: %s \n", managedCluster.GetName())
+
+	/* cluster := &clusterv1.ManagedCluster{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(mc.UnstructuredContent(), &cluster)
+	if err != nil {
+		klog.Fatalf("Error unmarshal managed cluster object%v \n", err)
+	} */
+
+	//TODO:
+	//get the actual values as mentioned here:
+	//https://github.com/open-cluster-management/perf-analysis/blob/master/Big%20Picture.md#acm-20-telemetry-data
+	managedClusterMetric.WithLabelValues(managedCluster.GetName(), "cluster_id", "type", "version", managedCluster.GetLabels()["cloud"], hubID).Set(1)
```

Comment:
Let's get all the mechanics in place and merged with this line and the other set under 
```
if false{
  // do it
```

That way we can get everything else merged

Reply:
Did not exactly follow this one. Let me know please.